### ISSUE TITLE
Super expression must either be null or a function, not undefined

### ISSUE THREAD

**Issue by MyGuySi** [babel/babel#6270]
I'm using React Native and getting the error in the title. The error is pointing at `actions.js:7` (see screenshot below) which is an import statement. It's importing a singleton of a class that extends a base class. I'm guessing that it's the `super()` call in the singleton class that's causing the issue.

It came up after restructuring the project and moving the files into new directories. The contents haven't changed, just the paths so I've triple checked the import paths and they're correct. I'd assume it would be a different error anyway if the path was incorrect. I'm having problems pin pointing where issue is here as I believe my import/export and class extension syntax is correct.

**Update**: `BaseApi` is `undefined` if I log it after importing in `api.js`

 Here's the error:

![screen shot 2017-09-19 at 09 56 37](https://user-images.githubusercontent.com/12952977/30584227-0778f09c-9d21-11e7-9025-b73d489d4adc.png)

### Input Code

```js
// actions.js
import { api } from "./api"; <-- this line is throwing the error
```

```js
// api.js
import BaseApi from "../baseApi";
    
export default class Api extends BaseApi {
  constructor() {
    super();
    ...
  }

  ...
}

export const api = new Api();
```

```js
// baseApi.js
export default class BaseApi {
  constructor() {
    ...
  }
  ...
}
```

### Babel Configuration

```js
// babel.rc
{
    "presets": [
        "react-native",
        "flow"
    ],
    "plugins": [
        "transform-inline-environment-variables",
        "transform-class-properties"
    ]
}
```

```js
// package.json
{
  "dependencies": {
    "babel-preset-expo": "^1.0.1",
    "react": "16.0.0-alpha.12",
    "react-native": "0.46.3",
    ...
  },
  "devDependencies": {
    "babel-eslint": "^7.1.1",
    "babel-jest": "18.0.0",
    "babel-plugin-transform-class-properties": "^6.23.0",
    "babel-plugin-transform-flow-strip-types": "^6.22.0",
    "babel-plugin-transform-inline-environment-variables": "^0.2.0",
    "babel-preset-es2015": "^6.22.0",
    "babel-preset-flow": "^6.23.0",
    "babel-preset-react-native": "1.9.1",
    ...
  },
  ...
}
```

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  Don't know, using React Native
| node             | 6.9.1
| npm              | 3.10.8
| Operating System | macOS Sierra 10.12.6

**Comment by MyGuySi** [comment]
Deleted my local repo, cloned again and it works fine... weird

**Comment by loganfsmyth** [comment]
Yeah this is generally how cycles manifest in Babel code right now. There's any number of reasons that `undefined` could get passed to a class, but that's the main reason it happens when it clearly seems like you imported it right.

I'm not sure there's anything specifically actionable to do that should make us leave this open though.
